src/gauging.jl

```diff
@@ -73,98 +78,66 @@ function vidal_gauge(
   return ψ_vidal, bond_tensors
 end
 
-"""Use an ITensorNetwork ψ in the symmetric gauge and its mts to put ψ into the vidal gauge. Return the bond tensors and ψ_vidal."""
 function vidal_gauge(
   ψ::ITensorNetwork,
-  pψψ::PartitionedGraph,
-  mts;
-  eigen_message_tensor_cutoff=10 * eps(real(scalartype(ψ))),
-  regularization=10 * eps(real(scalartype(ψ))),
-  edges=NamedGraphs.edges(ψ),
-  svd_kwargs...,
+  bond_tensors;
+  (cache!)=nothing,
+  cache_update_kwargs=default_cache_update_kwargs(cache!),
+  kwargs...,
 )
-  bond_tensors = initialize_bond_tensors(ψ)
-  return vidal_gauge(
-    ψ,
-    pψψ,
-    mts,
-    bond_tensors;
-    eigen_message_tensor_cutoff,
-    regularization,
-    edges,
-    svd_kwargs...,
-  )
-end
+  if isnothing(cache!)
+    cache! = Ref(default_cache(ψ))
+  end
 
-"""Put an ITensorNetwork into the vidal gauge (by computing the message tensors), return the network and the bond tensors. Will also return the mts that were constructed"""
-function vidal_gauge(
-  ψ::ITensorNetwork;
-  eigen_message_tensor_cutoff=10 * eps(real(scalartype(ψ))),
-  regularization=10 * eps(real(scalartype(ψ))),
-  niters=30,
-  target_canonicalness::Union{Nothing,Float64}=nothing,
-  verbose=false,
-  svd_kwargs...,
-)
-  ψψ = norm_network(ψ)
-  pψψ = PartitionedGraph(ψψ, group(v -> v[1], vertices(ψψ)))
+  cache![] = update(cache![]; cache_update_kwargs...)
+
+  return vidal_gauge(ψ, bond_tensors, cache![]; kwargs...)
+end
 
-  mts = belief_propagation(pψψ; niters, target_precision=target_canonicalness, verbose)
-  return vidal_gauge(
-    ψ, pψψ, mts; eigen_message_tensor_cutoff, regularization, svd_kwargs...
-  )
+"""Put an ITensorNetwork into the vidal gauge (by computing the message tensors), return the network and the bond tensors."""
+function vidal_gauge(ψ::ITensorNetwork; kwargs...)
+  bond_tensors = initialize_bond_tensors(ψ)
+  return vidal_gauge(ψ, bond_tensors; kwargs...)
 end
 
 """Transform from an ITensor in the Vidal Gauge (bond tensors) to the Symmetric Gauge (partitionedgraph, message tensors)"""
 function vidal_to_symmetric_gauge(ψ::ITensorNetwork, bond_tensors::DataGraph)
```

Comment:
Maybe we can just call this `symmetric_gauge`, and assume that if you pass both an `ITensorNetwork` and a `DataGraph` then it is in the Vidal gauge.

Reply:
Good point, I think the same applies to `symmetric_to_vidal_gauge` -> `vidal_gauge` as well